cleanlab/regression/rank.py

```diff
@@ -0,0 +1,150 @@
+            Please choose a valid scoring technique: {scoring_funcs.keys()}.
+            """
+        )
+
+    # Calculate scores
+    label_quality_scores = scoring_func(labels, predictions)
+    return label_quality_scores
+
+
+def get_residual_score_for_each_label(
+    labels: np.ndarray,
+    predictions: np.ndarray,
+) -> np.ndarray:
+    """Returns the residual based label-quality scores for each datapoints.
+
+    This is function to compute label-quality scores for regression datasets,
+    where lower score indicate labels less likely to be correct.
+
+    Residual based scores can work better for datasets where independent variables
+    are based out of normal distribution.
+
+    Parameters
+    ----------
+    labels: np.ndarray
+        Labels in the same format expected by the :py:func:`get_label_quality_scores <cleanlab.regression.rank.get_label_quality_scores>` function.
+
+    predictions: np.ndarray
+        Predicted labels in the same format expected by the :py:func:`get_label_quality_scores <cleanlab.regression.rank.get_label_quality_scores>` function.
+
+    Returns
+    -------
+    label_quality_scores: np.ndarray
+        Contains one score (between 0 and 1) per example.
+        Lower scores indicate more likely mislabled examples.
+
+    """
+    residual = predictions - labels
+    label_quality_scores = np.exp(-abs(residual))
+    return label_quality_scores
+
+
+# TODO - change name of the function
+# TODO - change name of function in test
+def get_score_to_named_for_each_label(
+    labels: np.ndarray,
+    predictions: np.ndarray,
+    *,
+    variance: float = 10,
+) -> np.ndarray:
+    """Returns label-quality scores.
+
+    This function computes label-quality scores for regression datasets,
+    where a lower score indicates labels that are less likely to be correct.
+
+    Parameters
+    ----------
+    labels: np.ndarray
+        Labels in the same format as expected by the :py:func:`get_label_quality_scores <cleanlab.regression.rank.get_label_quality_scores>` function.
+
+    predictions: np.ndarray
+        Predicted labels in the same format as expected by the :py:func:`get_label_quality_scores <cleanlab.regression.rank.get_label_quality_scores>` function.
+
+    variance: float, default = 10
+        Manipulates variance of the distribution of residual.
+
+    Returns
+    -------
+    label_quality_scores: np.ndarray
+        Contains one score (between 0 and 1) per example.
+        Lower scores indicate more likely mislabled examples.
+    """
+    residual = predictions - labels
+    labels = (labels - labels.mean()) / labels.std()
+    residual = np.sqrt(variance) * ((residual - residual.mean()) / residual.std())
+
+    # 2D features by combining labels and residual
+    features = np.array([labels, residual]).T
+
+    neighbors = int(np.ceil(0.1 * labels.shape[0]))
+    knn = NearestNeighbors(n_neighbors=neighbors, metric="euclidean").fit(features)
```

Comment:
this function should have another argument: `neighbor_metric` which is "euclidean" by default and used here

Reply:
resolved here [e9468dd](https://github.com/cleanlab/cleanlab/pull/572/commits/e9468dd9e476f9821e3c0ca2e4e17b00cb02d6a5)